### PR TITLE
expiry: prevent expired tasks from retrying automatically

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes https://github.com/cylc/cylc-flow/issues/6284

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/integration/test_task_pool.py [hunk 1]
```diff
@@ -2150,3 +2150,51 @@ async def test_trigger_unqueued(flow, scheduler, start):
         assert not schd.pool.task_queue_mgr.force_released, (
             "Triggering an unqueued task should not affect the force_released list"
         )
+
+
+@pytest.mark.parametrize('expire_type', ['clock-expire', 'manual'])
```
Comment (oliver-sanders) [inline tests/integration/test_task_pool.py:2155]: Manual expiry wasn't broken, but I included it in this test for completeness.

File: cylc/flow/task_pool.py [hunk 2]
```diff
@@ -2239,6 +2239,7 @@ def clock_expire_tasks(self):
                 # check if this task is clock expired
                 and itask.clock_expire()
             ):
+                self.task_queue_mgr.remove_task(itask)
```
Comment (hjoliver) [inline cylc/flow/task_pool.py:2242]: Damn, love an easy fix!

**Review by hjoliver — APPROVED** [review]

**Comment by hjoliver** [comment]
One review is sufficient for this.

**Review by wxtim — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Tests passed, pressing the button.

**Pull request merged**